src/docfx/lib/JsonUtility.cs

```diff
@@ -205,8 +208,58 @@ protected override JsonProperty CreateProperty(MemberInfo member, MemberSerializ
                         prop.Writable = true;
                     }
                 }
+
+                if (converter != null)
+                {
+                    if (prop.PropertyType.IsArray)
+                    {
+                        prop.ItemConverter = converter;
+                    }
+                    else
+                    {
+                        prop.Converter = converter;
+                    }
+                }
                 return prop;
             }
+
+            private static SchemaValidationConverter GetConverter(MemberInfo member)
+            {
+                var validators = member.GetCustomAttributes<ValidationAttribute>();
```

Comment:
>GetCustomAttributes [](start = 40, length = 19)

this is reflection, will it impact perf greatly?

Reply:
Tried to cache validators using MemberInfo as key, seems like equality of MemberInfo is not a trivial operation either. Update the reflection to not look into inherit types, will get back to this if we have perf issue later

---
In reply to: [206377654](https://github.com/dotnet/docfx/pull/3130#discussion_r206377654) [](ancestors = 206377654)

Reply:
sure, that's good enough to know at this moment. :)

---
In reply to: [206384045](https://github.com/dotnet/docfx/pull/3130#discussion_r206384045) [](ancestors = 206384045,206377654)